src/Interpreters/ThreadStatusExt.cpp

```diff
@@ -233,7 +233,8 @@ void ThreadStatus::attachToGroupImpl(const ThreadGroupPtr & thread_group_)
 {
     /// Attach or init current thread to thread group and copy useful information from it
     thread_group = thread_group_;
-    thread_group->linkThread(thread_id);
+    if (!internal_thread)
+        thread_group->linkThread(thread_id);
```

Comment:
A comment here would be good to have, e.g. why we now only do it only if !internal_thread

Reply:
If a change from other commit, it already in master.
https://github.com/ClickHouse/ClickHouse/pull/66230